pylearn2/datasets/dense_design_matrix.py

```diff
@@ -264,13 +264,14 @@ def _check_labels(self):
             assert self.y.ndim <= 2
             assert np.all(self.y < self.y_labels)
 
-    @functools.wraps(Dataset.iterator)
+    # Cannot wrap the iterator method cause it is abstract
+    # @functools.wraps(Dataset.iterator)
```

Comment:
Ok. 
These could be good starting points to address the issue:
http://code.activestate.com/recipes/576862-docstring-inheritance-decorator/
http://code.activestate.com/recipes/578587-inherit-method-docstrings-without-breaking-decorat/